save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/LnkContestProjectController.kt

```diff
@@ -153,6 +159,82 @@ class LnkContestProjectController(
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully fetched latest project execution in contest.")
+    @ApiResponse(responseCode = "404", description = "Either contest is not found or project is not found or execution is not found.")
+    fun getContestExecutionsForProject(
+        @PathVariable contestName: String,
+        @PathVariable organizationName: String,
+        @PathVariable projectName: String,
+        authentication: Authentication,
+    ): Flux<ExecutionDto> = Mono.justOrEmpty(contestService.findByName(contestName))
+        .switchIfEmptyToNotFound {
+            "Could not find contest with name $contestName."
+        }
+        .zipWith(projectService.findByNameAndOrganizationName(projectName, organizationName).toMono())
+        .switchIfEmptyToNotFound {
+            "Could not find project with name $organizationName/$projectName."
+        }
+        .flatMapMany { (contest, project) ->
+            lnkContestExecutionService.getPageExecutionsByContestAndProject(
+                contest,
+                project,
+                PageRequest.ofSize(MAX_AMOUNT)
+            ).toFlux()
+        }
+        .map {
+            it.execution.toDto()
+        }
+
+    @GetMapping("/{contestName}/executions/{organizationName}/{projectName}/latest")
+    @RequiresAuthorizationSourceHeader
+    @PreAuthorize("isAuthenticated()")
+    @Operation(
+        method = "GET",
+        summary = "Get latest project execution in contest.",
+        description = "Get latest execution of a project with given name in contest with given name.",
+    )
+    @Parameters(
+        Parameter(name = "contestName", `in` = ParameterIn.PATH, description = "name of an organization", required = true),
+        Parameter(name = "organizationName", `in` = ParameterIn.PATH, description = "name of an organization", required = true),
+        Parameter(name = "projectName", `in` = ParameterIn.PATH, description = "name of a project", required = true),
+    )
+    @ApiResponse(responseCode = "200", description = "Successfully fetched latest project execution in contest.")
+    @ApiResponse(responseCode = "404", description = "Either contest is not found or project is not found or execution is not found.")
+    fun getLatestExecutionOfProjectInContest(
+        @PathVariable organizationName: String,
+        @PathVariable projectName: String,
+        @PathVariable contestName: String,
+        authentication: Authentication
+    ): Mono<ExecutionDto> = Mono.justOrEmpty(contestService.findByName(contestName))
+        .switchIfEmptyToNotFound {
+            "Could not find contest with name $contestName."
+        }
+        .zipWith(projectService.findByNameAndOrganizationName(projectName, organizationName).toMono())
+        .switchIfEmptyToNotFound {
+            "Could not find project with name $organizationName/$projectName."
+        }
+        .flatMap { (contest, project) ->
```

Comment:
This part can be extracted and shared between the two new endpoints

Reply:
Oh sure, will fix it

Reply:
Done